code/site/components/com_pages/dispatcher/router/resolver/http.php

```diff
@@ -50,4 +50,27 @@ public function resolve(ComPagesDispatcherRouterInterface $router)
 
         return $route;
     }
+
+    /**
+     * Reversed routing
+     *
+     * Generate the URL for a route. Replace regexes with supplied parameters
+     *
+     * @param string $path The path to generate a route for
+     * @param array @params Associative array of parameters to replace placeholders with.
+     * @return false|KHttpUrl Returns the generated route
+     */
+    public function generate($path, array $query, ComPagesDispatcherRouterInterface $router)
+    {
+        $parts = explode('#', $path);
```

Comment:
@johanjanssens I think making this `$parts = explode('#', $path, 2);` would be safer against typos and wrong calls. Thoughts?